akka-stream-tests/src/test/scala/akka/stream/scaladsl/MapWithResourceSpec.scala

```diff
@@ -236,16 +236,20 @@ class MapWithResourceSpec extends StreamSpec(UnboundedMailboxConfig) {
     }
 
     "fail when close throws exception" in {
-      val out = TestSubscriber.probe[String]()
-      Source
-        .single(1)
+      val (pub, sub) = TestSource
+        .probe[Int]
         .mapWithResource(() => Iterator("a"))((it, _) => if (it.hasNext) Some(it.next()) else None, _ => throw TE(""))
         .collect { case Some(line) => line }
-        .runWith(Sink.fromSubscriber(out))
+        .toMat(TestSink.probe)(Keep.both)
+        .run()
 
-      out.request(61)
-      out.expectNext("a")
-      out.expectError(TE(""))
+      pub.ensureSubscription()
+      sub.ensureSubscription()
+      sub.request(1)
+      pub.sendNext(1)
+      sub.expectNext("a")
+      pub.sendComplete()
+      sub.expectError(TE(""))
```

Comment:
According to the RS specification exceptions can overtake elements so I'm not sure this is completely safe. Might work if there are no async boundaries but not great to assume that.

Perhaps the test can be written with `sub.expectNextOrError`?

Reply:
Thanks for the input, but here the complete is triggered by explicit send and the error only be trigged int he `onClose` so that should be safe?

Reply:
ok, yes that should work